### PR TITLE
Uniformize mesh editing tool with vector vertex tool

### PR DESCRIPTION
**Pull request by vcloarec**
This is the first part of an homogenization of the mesh editing tools with what is existing in the vector layer size. This work can be seen as a UI bug fixing.

Here, it make sense to have the button to enter in mesh element editing tool to be homogenized with the vertex tool button. So this button is moved from the "Mesh Digitizing Toolbar to the common "Digitizing Toolbar". 
The tool is renamed with "Mesh Element Tool" instead of "Digitize Mesh Element", 
and the tool bar is renamed with "Advanced Mesh Editing Toolbar" instead of "Mesh Digitizing Toolbar". Indeed, this tool bar will be only destined to welcome advanced editing tool as "Triangulation", "Force by Line", "Transform By Expression", ...

So, the next move after this PR is to move the buttons of the selection tools for mesh to the vector layer button for selection, the aim is to have the selection tools for mesh that replace the selection tools for vector layer when a mesh is the active layer.

For information, it was also planned to allow the user to use the "Delete Selected" button (the one used to delete selected vector layer features) to also delete mesh layer vertices or faces. But, I give up after some difficulties appear during implementation, essentially to provides three delete mode ("delete vertices filling hole", "delete vertices without filling hole", "delete faces") that are not always enables at the same time, with only one button, furthermore, that have only one action with vector layer.

ping @DelazJ 

**Before:**
image

**With this PR:**
editMeshElement

### CODE REVIEW

**Comment by vcloarec** [comment]
@DelazJ @nyalldawson @nirvn  any opinion for this ? I prefer to ask you before continuing with the buttons associated with the selections.

**Comment by uclaros** [comment]
This looks like a good move, the two tools are similar enough to share a button position!

Only slight drawback is that in the edge case that Vertex tool (all layers) is selected and a mesh layer activated, the tool will still be functional but will no longer have a visible button indicating that it is active.

I believe a similar approach for the two selection tools would also be nice. Will you continue on this or a separate PR @vcloarec ?

File: src/ui/qgisapp.ui [hunk 1]
```diff
@@ -3375,6 +3375,7 @@ Shows placeholders for labels which could not be placed, e.g. due to overlaps wi
  </widget>
  <resources>
   <include location="../../images/images.qrc"/>
+  <include location="../../images/images.qrc"/>
```
Comment (uclaros) [inline src/ui/qgisapp.ui:3378]:
```suggestion
  <include location="../../images/images.qrc"/>
```
qt designer hick-up?

**Comment by vcloarec** [comment]
> Will you continue on this or a separate PR @vcloarec ?

@uclaros, yes, it is the intention! But off course, if the community is OK with that, despite the issue you pointed.

**Comment by DelazJ** [comment]
@vcloarec sorry, this totally slipped my mind.
> But, I give up after some difficulties appear during implementation, essentially to provides three delete mode ("delete vertices filling hole", "delete vertices without filling hole", "delete faces") that are not always enables at the same time, with only one button, furthermore, that have only one action with vector layer.

What about a single button that depending on whether I press Shift or Ctrl while clicking will do one or another action? I don't know if we have this three states button. I can only think of the vertex tool [batch selecting mode](https://docs.qgis.org/3.22/en/docs/user_manual/working_with_vector/editing_geometry_attributes.html#basic-operations) which selects differently with (Ctrl) Shift+R. My concern is that the delete tools get hidden in a context menu despite erasing is a common editing tool (and they already need a complex keys combo)

**Comment by uclaros** [comment]
Hi @vcloarec, I gave this _uniformizing_ concept much thought and some chat with @wonder-sk  and came to the conclusion that it's not the best idea after all. Let me explain some key parts of my thoughts:
- Vertex tool (all layers) may be active and usable even if current layer has changed to a mesh layer
- Mesh layers select tools behave significantly different to vector layers' selection tools.
  - For vector layers a selection is not tied to editing mode, we select features to analyze and/or modify using distinct tools (move/rotate etc)
  - For mesh layers, selection acts like the _Vertex tool_ selections, which do not have a separate action icon, they're part of the mesh digitizing tool and are not usable when not in edit mode
-  _Vertex tool_ selection modes are different to mesh selection modes
- - Vertex tool selects vertices. In order to get the equivalent for mesh one needs to hold _alt_
- - Vertex tool has built in select by polygon with _alt_, mesh needs a separate tool for that
- - Vertex tool has range selection


Given those differences (I might be forgetting more) I think it would be best to keep layer editing separate judging by layer type and not mix mesh layer editing with vector layer editing.

**_I would therefore propose the following approach:_**
- Add the toggle editing icon to the mesh toolbar and keep all mesh related actions grouped on the mesh toolbar. Anyone wanting to edit meshes will have all the available actions nicely grouped in a single place knowing that all the tools he can use are in this toolbar.
- Make sure that the mesh toolbar is shown when starting editing on a mesh layer.
- Keep the _show Z value editor_ checkbox, this is great addition and helps keeping the ui clean.
- Add all future mesh editing tools to that toolbar (maybe tools for actions now accessible only in the context menu, like triangulate, deletes etc)

Hope the above make sense!

**Comment by uclaros** [comment]
Re-pinging @DelazJ @nyalldawson @nirvn if you have any opinion on this regarding my latest suggestion.

**Pull request closed without merging**